watertap/core/zero_order_costing.py

```diff
@@ -0,0 +1,510 @@
+        basis as a unit of that type is costed.
+        """
+        # Load case study definition from file
+        cs_def = _load_case_study_definition(self)
+
+        # Register currency and conversion rates
+        if "currency_definitions" in cs_def:
+            pyo.units.load_definitions_from_strings(
+                cs_def["currency_definitions"])
+        else:
+            register_idaes_currency_units()
+
+        # Set the base year for all costs
+        self.base_currency = getattr(pyo.units, cs_def["base_currency"])
+        # Set a base period for all operating costs
+        self.base_period = getattr(pyo.units, cs_def["base_period"])
+
+        # Define expected flows
+        self.defined_flows = {}
+        for f, v in cs_def["defined_flows"].items():
+            self.defined_flows[f] = v["value"]*getattr(pyo.units, v["units"])
+
+        # Costing factors
+        self.plant_lifetime = pyo.Var(initialize=30,
+                                      units=self.base_period,
+                                      doc="Plant lifetime")
+        self.utilization_factor = pyo.Var(
+            initialize=1,
+            units=pyo.units.dimensionless,
+            doc='Plant capacity utilization [%]')
+
+        self.land_cost_percent_FCI = pyo.Var(initialize=0.07,
+                                             units=pyo.units.dimensionless,
+                                             doc="Land cost as % FCI")
+        self.working_capital_percent_FCI = pyo.Var(
+            initialize=0.008,
+            units=pyo.units.dimensionless,
+            doc="Working capital as % FCI")
+        self.salaries_percent_FCI = pyo.Var(
+            initialize=0.07,
+            units=1/self.base_period,
+            doc="Salaries as % FCI")
+        self.benefit_percent_of_salary = pyo.Var(
+            initialize=0.07,
+            units=pyo.units.dimensionless,
+            doc="Benefits as % salaries")
+        self.maintenance_costs_percent_FCI = pyo.Var(
+            initialize=0.07,
+            units=1/self.base_period,
+            doc="Maintenance and contingency costs as % FCI")
+        self.laboratory_fees_percent_FCI = pyo.Var(
+            initialize=0.07,
+            units=1/self.base_period,
+            doc="Laboratory fees as % FCI")
+        self.insurance_and_taxes_percent_FCI = pyo.Var(
+            initialize=0.07,
+            units=1/self.base_period,
+            doc="Insurance and taxes as % FCI")
+
+        self.wacc = pyo.Var(initialize=0.05,
+                            units=pyo.units.dimensionless,
+                            doc='Weighted Average Cost of Capital (WACC)')
+        self.capital_recovery_factor = pyo.Expression(
+            expr=((self.wacc *
+                   (1 + self.wacc)**(self.plant_lifetime/self.base_period)) /
+                  (((1 + self.wacc)**(self.plant_lifetime/self.base_period)) -
+                   1) / self.base_period))
```

Comment:
``base_period`` isn't in the WT3 formula for ``capital_recovery_factor`` - what is it for?